### DISCUSSION TITLE
Is there a plan to enable CI/CD integration?

### DISCUSSION THREAD

**Discussion by ghost** [DaxStudio/DaxStudio#552] [Q&A]
We have a team here that is interested in DAX Studio for Power BI testing.  They have started a PoC and so far are having some success.

We would like to include their tests in our automated pipelines, but it looks like the tool is Desktop app only 😞.  If I am wrong happy to RTM if it is available or read source if you can point me to it... Even a CLI would work for us.  If there is no integration currently, is it an item for your roadmap?  

Thanks in advance

**Comment by dgosbell** [comment]
I don't see how DAX Studio would add a lot of value to a CI/CD pipeline, we don't have any sort of assertion framework that would help with automated test. If all you want to do is run queries you can do that with a small amount of Powershell as in [this gist](https://gist.github.com/dgosbell/75a4c2b6c1eae6ba1a8f4fdddd995119) which executes a dax query and outputs the results to a csv.

Personally I am using [nbi.io](https://nbi.io) for our automated testing. It has a huge array of features specifically targeted at automated testing of BI solutions.
